src/modules/facility-loan/facility-loan.service.test.ts

```diff
@@ -0,0 +1,58 @@
+import { PROPERTIES } from '@ukef/constants';
+import { AcbsFacilityLoanService } from '@ukef/modules/acbs/acbs-facility-loan.service';
+import { AcbsAuthenticationService } from '@ukef/modules/acbs-authentication/acbs-authentication.service';
+import { DateStringTransformations } from '@ukef/modules/date/date-string.transformations';
+import { getMockAcbsAuthenticationService } from '@ukef-test/support/abcs-authentication.service.mock';
+import { GetFacilityLoanGenerator } from '@ukef-test/support/generator/get-facility-loan-generator';
+import { RandomValueGenerator } from '@ukef-test/support/generator/random-value-generator';
+import { when } from 'jest-when';
+
+import { FacilityLoanService } from './facility-loan.service';
+
+describe('FacilityLoanService', () => {
+  const portfolioIdentifier = PROPERTIES.GLOBAL.portfolioIdentifier;
+  const valueGenerator = new RandomValueGenerator();
+  const idToken = valueGenerator.string();
+  const facilityIdentifier = valueGenerator.facilityId();
+
+  const { facilityLoansFromApi: expectedFacilityLoans, facilityLoansInAcbs } = new GetFacilityLoanGenerator(
+    valueGenerator,
+    new DateStringTransformations(),
+  ).generate({ numberToGenerate: 2, facilityIdentifier, portfolioIdentifier });
+
+  let acbsAuthenticationService: AcbsAuthenticationService;
+  let service: FacilityLoanService;
+
+  let getFacilityLoansAcbsService: jest.Mock;
+
+  beforeEach(() => {
+    const mockAcbsAuthenticationService = getMockAcbsAuthenticationService();
+    acbsAuthenticationService = mockAcbsAuthenticationService.service;
+    const acbsAuthenticationServiceGetIdToken = mockAcbsAuthenticationService.getIdToken;
+    when(acbsAuthenticationServiceGetIdToken).calledWith().mockResolvedValueOnce(idToken);
+
+    const acbsService = new AcbsFacilityLoanService(null, null);
+    getFacilityLoansAcbsService = jest.fn();
+    acbsService.getLoansForFacility = getFacilityLoansAcbsService;
+
+    service = new FacilityLoanService(acbsAuthenticationService, acbsService, new DateStringTransformations());
+  });
+
+  describe('getLoansForFacility', () => {
+    it('returns a transformation of the loans from ACBS', async () => {
+      when(getFacilityLoansAcbsService).calledWith(portfolioIdentifier, facilityIdentifier, idToken).mockResolvedValueOnce(facilityLoansInAcbs);
+
+      const loans = await service.getLoansForFacility(facilityIdentifier);
+
+      expect(loans).toStrictEqual(expectedFacilityLoans);
+    });
+
+    it('returns an empty array if ACBS returns an empty array', async () => {
+      when(getFacilityLoansAcbsService).calledWith(portfolioIdentifier, facilityIdentifier, idToken).mockResolvedValueOnce([]);
```

Comment:
404 assertion please?

Reply:
I don't think a 404 assertion makes sense here, as a 404 happens if the ACBS service throws a  `AcbsResourceNotFoundException` (which then gets mapped to a 404 by the interceptor). There's already an assertion that the ACBS service throws a  `AcbsResourceNotFoundException` if ACBS returns a 400 'Facility not found' response, and an API test to check this gets mapped to a 404.